homeassistant/components/switcher_kis/__init__.py

```diff
@@ -0,0 +1,268 @@
+Please follow configuring instructions here:
+    https://www.home-assistant.io/components/switcher_kis/
+
+Author: Tomer Figenblat
+"""
+
+from asyncio import TimeoutError as Asyncio_TimeoutError, wait_for
+from datetime import datetime, timedelta
+from functools import partial
+from logging import getLogger
+from typing import Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.switch import DOMAIN as SWITCH_DOMAIN
+from homeassistant.const import CONF_ICON, CONF_NAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.core import Event, HomeAssistant, ServiceCall
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.discovery import (async_listen_platform,
+                                             async_load_platform)
+from homeassistant.helpers.event import async_call_later
+
+_LOGGER = getLogger(__name__)
+
+REQUIREMENTS = ['aioswitcher==2019.3.21']
+DOMAIN = 'switcher_kis'
+ENTITY_ID_FORMAT = DOMAIN + '_{}'
+
+CONF_AUTO_OFF = 'auto_off'
+CONF_DAYS = 'days'
+CONF_DEVICE_ID = 'device_id'
+CONF_DEVICE_PASSWORD = 'device_password'
+CONF_END_TIME = 'end_time'
+CONF_INCLUDE_SCHEDULE_SENSORS = 'include_schedule_sensors'
+CONF_PHONE_ID = 'phone_id'
+CONF_RECURRING = 'recurring'
+CONF_SCHEDULE_ID = 'schedule_id'
+CONF_SCHEDULE_SCAN_INTERVAL = 'schedules_scan_interval'
+CONF_START_TIME = 'start_time'
+
+DEFAULT_NAME = 'boiler'
+DEFAULT_SCHEDULES_SCAN_INTERVAL = timedelta(minutes=5)
+
+SWITCHER_MAX_SCHEDULES = 8
+
+DISCOVERY_CONFIG = 'config'
+DISCOVERY_DEVICE = 'device'
+DISCOVERY_IP_ADDRESS = 'ip_address'
+DISCOVERY_SCHEDULES = 'schedules'
+
+EVENT_SWITCHER_DEVICE_UPDATED = 'switcher_device_update'
+UPDATED_DEVICE = 'updated_device'
+
+STATE_NOT_CONFIGURED = "Not configured"
+STATE_NOT_ENABLED = "Not enabled"
+STATE_NOT_RUNNING = "Not running"
+
+ATTR_AUTO_OFF_SET = 'auto_off_set'
+ATTR_DAYS = 'days'
+ATTR_DEVICE_NAME = 'device_name'
+ATTR_DURATION = 'duration'
+ATTR_ELECTRIC_CURRNET = 'electric_current'
+ATTR_ENABLED = 'enabled'
+ATTR_END_TIME = 'end_time'
+ATTR_IP_ADDRESS = 'ip_address'
+ATTR_LAST_DATA_UPDATE = 'last_data_update'
+ATTR_LAST_STATE_CHANGE = 'last_state_change'
+ATTR_RECURRING = 'recurring'
+ATTR_REMAINING_TIME = 'remaining_time'
+ATTR_SCHEDULE_ID = 'schedule_id'
+ATTR_START_TIME = 'start_time'
+
+SERVICE_CREATE_SCHEDULE = 'create_schedule'
+SERVICE_DELETE_SCHEDULE = 'delete_schedule'
+SERVICE_DISABLE_SCHEDULE = 'disable_schedule'
+SERVICE_ENABLE_SCHEDULE = 'enable_schedule'
+SERVICE_SET_AUTO_OFF = 'set_auto_off'
+SERVICE_TURN_ON_15 = 'turn_on_15_minutes'
+SERVICE_TURN_ON_30 = 'turn_on_30_minutes'
+SERVICE_TURN_ON_45 = 'turn_on_45_minutes'
+SERVICE_TURN_ON_60 = 'turn_on_60_minutes'
+SERVICE_UPDATE_DEVICE_NAME = 'update_device_name'
+SERVICE_UPDATE_SCHEDULES = 'update_schedules'
+
+# Do not edit the DAY_% values, they must match the ones in aioswitcher.consts
+DAY_FRIDAY = "Friday"
```

Comment:
Why can't we get these from the aioswitcher.consts directly?

Reply:
@rohankapoorcom 
Those values are needed for the component's configuration schema which resides on the top level of the `__init__.py` file.
As I understand, HA's architecture doesn't allow importing third-party modules at this stage.

I would very much prefer to use aioswitcher.consts directly, I think it will be less error prone.
Please, if you can think of a way for me to accomplish that, I would be happy to make the adjustments.

Reply:
Removed, not relevant anymore.